client/src/main/java/org/schlunzis/kurtama/client/server/DockerServer.java

```diff
@@ -0,0 +1,132 @@
+package org.schlunzis.kurtama.client.server;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.schlunzis.kurtama.client.util.VersionManager;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.concurrent.Semaphore;
+
+@Slf4j
+class DockerServer extends Server {
+
+    private final Semaphore mutex = new Semaphore(1);
+    private final String composeString;
+    private String path;
+
+    DockerServer(VersionManager versionManager, LogSink logSink) {
+        super(logSink);
+        try {
+            String cs = IOUtils.resourceToString("/server/server-compose.yml", StandardCharsets.UTF_8);
+            this.composeString = cs.replace("{version}", versionManager.getDockerVersion());
+        } catch (IOException e) {
+            log.error("Error while reading docker-compose.yml", e);
+            throw new RuntimeException("Error while reading docker-compose.yml", e);
+        }
+    }
+
+    @Override
+    public boolean testRequirements() {
+        log.info("Testing for docker installation");
+        ProcessBuilder processBuilderDocker = new ProcessBuilder("docker", "--version");
+        ProcessBuilder processBuilderCompose = new ProcessBuilder("docker", "compose", "version");
+        try {
+            Process processDocker = processBuilderDocker.start();
+            StreamGobbler streamGobbler = new StreamGobbler(processDocker.getInputStream());
+            executor.submit(streamGobbler);
+            int exitCodeDocker = processDocker.waitFor();
+            String[] output = streamGobbler.getOutput();
+            Arrays.stream(output).forEach(log::info);
+
+            Process processCompose = processBuilderCompose.start();
+            streamGobbler = new StreamGobbler(processCompose.getInputStream());
+            executor.submit(streamGobbler);
+            int exitCodeCompose = processCompose.waitFor();
+            output = streamGobbler.getOutput();
+            Arrays.stream(output).forEach(log::info);
+
+            return exitCodeDocker == 0 && exitCodeCompose == 0;
+        } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
+            log.error("Error while testing DOCKER requirements", e);
+            return false;
+        } catch (IOException e) {
+            log.error("Error while testing DOCKER requirements", e);
+        }
+        return false;
+    }
+
+    @Override
+    public void run(int port, String path) {
+        mutex.acquireUninterruptibly();
+        setStatus(ServerStatus.DOWNLOADING);
+        String runConfig = composeString.replace("{port}", String.valueOf(port));
```

Comment:
```suggestion
        String runConfig = composeString.replace("{port}", Integer.toString(port));
```

Again, just showing that I know something as well...